packages/terra-clinical-data-grid/src/DataGrid.jsx

```diff
@@ -0,0 +1,1200 @@
+      return undefined;
+    }
+
+    return (
+      <HeaderCell
+        key={columnId}
+        columnId={columnId}
+        text={columnData.text}
+        sortIndicator={columnData.sortIndicator}
+        width={`${DataGrid.getWidthForColumn(columnData, defaultColumnWidth)}px`}
+        isSelectable={columnData.isSelectable}
+        isResizable={hasResizableColumns && columnData.isResizable}
+        onResizeEnd={this.updateColumnWidth}
+        onSelect={onColumnSelect}
+        selectableRefCallback={(ref) => { this.headerCellRefs[columnId] = ref; }}
+      >
+        {columnData.component}
+      </HeaderCell>
+    );
+  }
+
+  renderHeaderRow() {
+    const {
+      headerHeight,
+    } = this.props;
+
+    const {
+      pinnedColumnWidth,
+      overflowColumnWidth,
+    } = this.state;
+
+    return (
+      <div
+        className={cx('header-container')}
+        style={this.generateHeaderContainerStyle(headerHeight)}
+      >
+        <div
+          className={cx('pinned-header')}
+          style={this.generatePinnedContainerStyle(pinnedColumnWidth)}
+        >
+          {DataGrid.getPinnedColumns(this.props).map(column => this.renderHeaderCell(column))}
+        </div>
+        <div
+          className={cx('header-overflow-container')}
+          ref={this.setHeaderOverflowContainerRef}
+          onScroll={this.synchronizeHeaderScroll}
+        >
+          <div
+            className={cx('overflow-header')}
+            style={this.generateOverflowContainerStyle(overflowColumnWidth, headerHeight)}
+          >
+            {DataGrid.getOverflowColumns(this.props).map(column => this.renderHeaderCell(column))}
+          </div>
+        </div>
+        <div
+          className={cx('header-scrollbar-buffer')}
+          ref={this.setHeaderScrollbarBufferRef}
+        />
+      </div>
+    );
+  }
+
+  renderSectionHeader(section, isPinned) {
+    const { onRequestSectionCollapse } = this.props;
+
+    const shouldRenderSectionHeaderContainer = section.isCollapsible || section.text || section.startAccessory || section.endAccessory || section.component;
+
+    return (
+      shouldRenderSectionHeaderContainer ? (
+        <div
+          key={section.id}
+          className={cx('section-header-container')}
+        >
+          { isPinned ? (
+            <SectionHeader
+              sectionId={section.id}
+              text={section.text}
+              startAccessory={section.startAccessory}
+              endAccessory={section.endAccessory}
+              isCollapsible={section.isCollapsible}
+              isCollapsed={section.isCollapsed}
+              onRequestSectionCollapse={onRequestSectionCollapse}
+              selectableRefCallback={(ref) => {
+                this.sectionRefs[section.id] = ref;
+              }}
+            >
+              {section.component}
+            </SectionHeader>
+          ) : null}
+        </div>
+      ) : null
+    );
+  }
+
+  renderRowSelectionCell(section, row, column) {
```

Comment:
RenderRowSelectionCell and RenderCell are very similar. Given the complexity of DataGrid and the simplicity of subcompenents/Cell, would it make sense to push this differentiation to the Cell component level? It might not makes to

Reply:
They are pretty close, but I don't think pushing the logic into the cell works. I instead made renderCell take more arguments so that renderRowSelectionCell can just provide overrides as it needs to.